ts/backend/src/app/features/utils/aggregator.spec.ts

```diff
@@ -0,0 +1,640 @@
+  {
+    scenario_id: '133636be-47f0-40ce-ac83-ce62ed72bbd8',
+    submission_id: '5ccc78a3-c60c-4b52-a995-000650edc49b',
+    key: 'penalty',
+    score: 5,
+  },
+  // submission "1", scenarios 5-6 left out to test NaN behavior
+  // submission "2", scenario "1"
+  {
+    scenario_id: 'ccf7c5f8-60d4-4bca-99e6-3a7abed22c37',
+    submission_id: '7a9b5af6-9604-49f7-9dc2-7be73d80b291',
+    key: 'score',
+    score: -2.0,
+  },
+  // submission "3", scenario "1"
+  {
+    scenario_id: 'c142fbd1-3e1d-4ca7-9a21-aeacc9d45ae9',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'score',
+    score: 1.0,
+  },
+  {
+    scenario_id: 'c142fbd1-3e1d-4ca7-9a21-aeacc9d45ae9',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'duration',
+    score: 2.0,
+  },
+  {
+    scenario_id: 'c142fbd1-3e1d-4ca7-9a21-aeacc9d45ae9',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'done',
+    score: 1.0,
+  },
+  // submission "3", scenario "2"
+  {
+    scenario_id: '4145ca69-d13f-426f-befd-20bd087383ee',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'score',
+    score: 2.0,
+  },
+  {
+    scenario_id: '4145ca69-d13f-426f-befd-20bd087383ee',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'duration',
+    score: 4.0,
+  },
+  {
+    scenario_id: '4145ca69-d13f-426f-befd-20bd087383ee',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'done',
+    score: 1.0,
+  },
+  // submission "3", scenario "3"
+  {
+    scenario_id: '654e402a-63a2-4d67-9578-5b13ec409d76',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'score',
+    score: 3.0,
+  },
+  {
+    scenario_id: '654e402a-63a2-4d67-9578-5b13ec409d76',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'duration',
+    score: 8.0,
+  },
+  {
+    scenario_id: '654e402a-63a2-4d67-9578-5b13ec409d76',
+    submission_id: '12ac0ba9-5e4a-4f4d-9b82-90ded100478f',
+    key: 'done',
+    score: 1.0,
+  },
+]
+
+/**
+ * Replaces resources of type `definition` by their ID during `JSON.stringify`.
+ * @param key Key of property being stringified.
+ * @param value Value of property being stringified.
+ * @returns Anything stringifyable.
+ */
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const ResourceAndScoreReplacer = (key: string, value: any) => {
+  if (typeof value === 'object') {
+    if (key === 'definition' && 'id' in value) {
+      return value.id
+    }
+  }
+  return value
+}
+
+describe('Aggregator', () => {
+  beforeAll(() => {
+    const benchmarkGroup = Aggregator.getBenchmarkGroupScored(
+      dummyBenchmarks,
+      dummyTests,
+      dummyScenarios,
+      dummySubmissions,
+      dummyResults,
+    )
+    console.log('Tree structure results:')
+    // console.log(JSON.stringify(benchmarkGroup))
+    console.log(JSON.stringify(benchmarkGroup, ResourceAndScoreReplacer))
+  })
+
+  it('should aggregate everything', () => {
```

Comment:
add functional testing values aggregation func,level, case
* for func in SUM, NANSUM etc.
* for levels in {scenario->test, test->leadeboard, test->campaign}
* for case in with/without nan values